cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -55,12 +56,18 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	mf := otelmetrics.NewFactory(s.telset.MeterProvider)
 	baseFactory := mf.Namespace(metrics.NSOptions{Name: "jaeger"})
 	queryMetricsFactory := baseFactory.Namespace(metrics.NSOptions{Name: "query"})
-	f, err := jaegerstorage.GetStorageFactory(s.config.Storage.TracesPrimary, host)
+
+	f, err := jaegerstorage.GetStorageFactoryV2(s.config.Storage.TracesPrimary, host)
 	if err != nil {
 		return fmt.Errorf("cannot find primary storage %s: %w", s.config.Storage.TracesPrimary, err)
 	}
 
-	spanReader, err := f.CreateSpanReader()
+	traceReader, err := f.CreateTraceReader()
+	if err != nil {
+		return fmt.Errorf("cannot create trace reader: %w", err)
+	}
+
+	spanReader, err := factoryadapter.GetV1Reader(traceReader)
```

Comment:
the idea was to push this conversion down into query service (as deep as possible), otherwise we can never make it work with v2 API

Reply:
@yurishkuro Oh okay I see. So a couple of follow-ups 
- Do we want to take the `traceReader` as part of the constructor, hold it in the `QueryService` struct and then perform the conversion before calling the underlying method (in https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/querysvc/query_service.go#L63 for example)?
- Do we want to use the storage v2 factory in v1 context as well? The factory is initialized differently in v1 (example: https://github.com/jaegertracing/jaeger/blob/main/cmd/query/main.go#L42).

Reply:
> Do we want to take the traceReader as part of the constructor, hold it in the QueryService struct and then perform the conversion before calling the underlying method 

yes

> Do we want to use the storage v2 factory in v1 context as well?

Not quite following, what do you mean by context?